backend/file-ucloud-service/src/jvmMain/kotlin/services/FileQueries.kt

```diff
@@ -148,3 +181,29 @@ class FileQueries(
         private const val DIR_CACHE_EXPIRATION = 1000L * 60 * 5
     }
 }
+
+fun sortFiles(
+    nativeFs: NativeFS,
+    sortBy: FilesSortBy,
+    sortOrder: SortDirection?,
+    foundFiles: List<InternalFile>,
+    foundFilesToStat: HashMap<String, NativeStat>
+): List<InternalFile> {
+    if (sortBy != FilesSortBy.PATH) foundFiles.forEach { foundFilesToStat[it.path] = nativeFs.stat(it) }
+    val pathComparator = compareBy(String.CASE_INSENSITIVE_ORDER, InternalFile::path)
+    var comparator = when (sortBy) {
+        FilesSortBy.PATH -> pathComparator
+        FilesSortBy.SIZE -> kotlin.Comparator<InternalFile> { a, b ->
+            ((foundFilesToStat[a.path]?.size ?: 0L) - (foundFilesToStat[b.path]?.size ?: 0L)).toInt()
+        }.thenComparing(pathComparator)
+        FilesSortBy.CREATED_AT -> {
```

Comment:
Created at doesn't exist for files. It is hard to believe, but this is not something that is standard in file systems.